include/pisa/scorer/scorer_factory.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "scorer.hpp"
+#include "ql.hpp"
+#include "bm25.hpp"
+
+namespace pisa {
+
+auto get_scorer = [](std::string const &scorer_name, auto const &wdata) -> std::unique_ptr<scorer<decltype(wdata)>> {
```

Comment:
I agree with you. Not sure it can be moved to the scorer file as it depends on `bm25.hpp` and `ql.hpp`. At the same time these classes depend on the `scorer.hpp` one since they derive from it.